crates/cli/src/state.rs

```diff
@@ -44,7 +45,7 @@ impl State {
             lockfile: call_load_lockfile(config.lockfile, Lockfile::load_from_current_dir)
                 .map_err(InitStateError::LoadLockfile)?,
             http_client: Client::new(),
-            tarball_mem_cache: MemCache::new(),
+            tarball_mem_cache: MemCache::new(HashMap::new()),
```

Comment:
I wonder if `Default::default()` would be possible?

```suggestion
            tarball_mem_cache: MemCache::default(),
```